e2e/test/scaling/provider.go

```diff
@@ -94,30 +113,25 @@ func (p *Provider) NumMasters() (int, error) {
 }
 
 func (p *Provider) NumWorkers() (int, error) {
-	customObject, err := p.hostFramework.G8sClient().ProviderV1alpha1().AzureConfigs("default").Get(p.clusterID, metav1.GetOptions{})
+	ctx := context.TODO()
+	machinePool, err := p.findMachinePool(ctx)
 	if err != nil {
 		return 0, microerror.Mask(err)
 	}
 
-	num := len(customObject.Spec.Azure.Workers)
-
-	return num, nil
+	return int(*machinePool.Spec.Replicas), nil
 }
 
 func (p *Provider) RemoveWorker() error {
-	patches := []Patch{
-		{
-			Op:   "remove",
-			Path: "/spec/azure/workers/1",
-		},
-	}
-
-	b, err := json.Marshal(patches)
+	ctx := context.TODO()
```

Comment:
```suggestion
	ctx := context.Background()
```
![Background](https://i.imgur.com/9tIuPBH.png)